docs/source/data/data_catalog.md

```diff
@@ -30,6 +30,41 @@ shuttles:
   load_args:
     engine: openpyxl # Use modern Excel engine (the default since Kedro 0.18.0)
 ```
+
+### Configuring dataset parameters in `catalog.yml`
+
+The dataset configuration in `catalog.yml` is defined as follows:
+1. The Top-level key is the dataset name used as a dataset identifier in the catalog - `shuttles`, `weather` in the example below.
+2. The next level includes multiple keys. The first one is the mandatory key, `type,` which defines the type of dataset to use.
+The rest of the keys are dataset properties and vary depending on the implementation.
```

Comment:
I think it would be cool to do something like

> [!IMPORTANT]  
> Kedro datasets make every intention to not make any assumptions and delegate any of the `load_args` / `save_args` directly to the underlying implementation.